pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -716,9 +713,6 @@ func validateNetworkFlag(data map[string]interface{}, create bool) error {
 			}
 			return fmt.Errorf("enableNetworkPolicy should be false for non-RKE clusters")
 		}
-		if plugin != "canal" {
```

Comment:
Do we not want to run validation so that if user enable PNI with flannel, we can raise an error?

Reply:
There are 3 options for Flannel:
1. Disable in UI and backend
2. Disable in UI and not backend
3. Allow in UI and backend

I've opted for `2` here per some conversations I've had around this. With this approach, it's still technically possible to enable PNI for Flannel if you manually set `enable_network_policy: true` in the RKE Config. Enabling NetworkPolicy for a CNI that doesn't support it shouldn't have any adverse effects, the policies just don't get enforced.

Reply:
UI Issue [is here](https://github.com/rancher/rancher/issues/32106) for reference @StrongMonkey